### ISSUE TITLE
MultiClass hangs

### ISSUE THREAD

**Issue by DGolubets** [catboost/catboost#130]
Hi,

I want to give Catboost a try, but can't make it run.

When I start fitting, it uses all 8 CPU cores to 100%, eats all available memory (60GB) and gets itself killed by Linux eventually.

The only line I see is:
```Borders for float features generated```

Code:
```python
    train_pool = Pool('data/datasets/train.tsv')
    test_pool = Pool('data/datasets/test.tsv')

    params = {
        'iterations':100, 
        'depth':2, 
        'learning_rate':0.01, 
        'loss_function':'MultiClass', 
        'verbose': True
        }
    model = CatBoost(params)
    model.fit(train_pool, eval_set = test_pool) 
```
Dataset is kinda small, few thousand rows of this format (first column is category id):
```
2766	0.9951810700532321	0.0001356243974014171	0.09805416571327646	-0.999113769235956	-0.04209128322017364	0.41818181818181815	0.2857142857142857	0.6007983831976773
2905	0.9973613087009463	-0.0009429794117380711	0.07259153322643554	-0.9232130421198814	-0.3842885359463043	0.41818181818181815	0.2857142857142857	0.6007983831976773
3856	0.99850004568635	-0.008682000039898276	0.054058132040093805	-0.8642493469339418	-0.5030636801879609	0.41818181818181815	0.2857142857142857	0.6007983831976773
...
```

**Comment by annaveronika** [comment]
what are the target values? is it the first column?

**Comment by DGolubets** [comment]
Yes

**Comment by annaveronika** [comment]
In multiclass CatBoost tries to estimate probabilities of each class, thus each leaf of a each tree contains a number for each resulting category. If you have many categories it becomes bottleneck to calculate these probabilities. Plus it is inefficient to use gradient boosting in this manner if you have thousands of classes. You could try reducing amount of classes or using other methods.

**Comment by DGolubets** [comment]
I just was about to write that I managed to make it work with ~50 classes on reduced dataset..
But in reality I have ~3k of them.

I didn't know about this limitation of boosting.
Thanks for explaining.
I guess will stick to neural networks.
